diff --git a/include/pocl_context.h b/include/pocl_context.h
index 3e125c18..7707ca04 100644
--- a/include/pocl_context.h
+++ b/include/pocl_context.h
@@ -67,8 +67,8 @@ struct pocl_context {
     __dst->local_size[0] = __src->local_size[0];			\
     __dst->local_size[1] = __src->local_size[1];			\
     __dst->local_size[2] = __src->local_size[2];			\
-    __dst->printf_buffer = __src->printf_buffer;			\
-    __dst->printf_buffer_position = __src->printf_buffer_position;	\
+    __dst->printf_buffer = (uint) __src->printf_buffer;			\
+    __dst->printf_buffer_position = (uint) __src->printf_buffer_position;	\
     __dst->printf_buffer_capacity = __src->printf_buffer_capacity;	\
   } while (0)
 
